### PR TITLE
Bump CAPI to v1.4.1

### LINKED ISSUE #4146: Test CAPI with release 1.4 RC
Release candidate for v1.4.0 release of CAPI is out: https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.4.0-rc.0

This issue is to track bumping the version of CAPI in CAPA and running all the tests with it to help validate the new release and ensure there are no regressions.

See [the Cluster-API book](https://main.cluster-api.sigs.k8s.io/developer/providers/v1.3-to-v1.4.html) for an overview of relevant changes between Cluster API v1.3 and v1.4 that may need changes in CAPA.

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind support

**What this PR does / why we need it**:
This PR bumps CAPI to v1.4.1, and resolves all the related errors.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4146 

**Special notes for your reviewer**:
Didnt add new CAPI kcp remediation test because of limitation to use it for provider. Details could be found [here](https://kubernetes.slack.com/archives/C8TSNPY4T/p1680525266510109). Raised the [issue](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4198) for the same.

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

**Release note**:

```release-note
Bump CAPI to v1.4.0
```

### CODE REVIEW

File: controllers/awsmachine_controller_unit_test.go [hunk 1]
```diff
@@ -1892,7 +1892,7 @@ func TestAWSMachineReconcilerAWSClusterToAWSMachines(t *testing.T) {
 			awsMachine: &clusterv1.Machine{
 				ObjectMeta: metav1.ObjectMeta{
 					Labels: map[string]string{
-						clusterv1.ClusterLabelName: "aws-test-2",
+						clusterv1.ClusterNameLabel: "aws-test-2",
```
[the remark this answers is not in the document]
Comment (Ankitasw) [reply controllers/awsmachine_controller_unit_test.go:1895]: 🤣

[the remark this answers is not in the document]
Comment (wyike) [reply controllers/awsmachine_controller_unit_test.go:1895]: `clusterv1.ClusterLabelName ===> clusterv1.ClusterNameLabel` Is this a change in CAPI 1.4?

**Comment by wyike** [comment]
Hi all, if downstream is going to bump CAPI 1.4 when it is GA, it looks CAPA needs some code change on it. How should downstream bump CAPA, if no new CAPA release comes out when CAPI 1.4 release GA?

**Comment by Ankitasw** [comment]
> Hi all, if downstream is going to bump CAPI 1.4 when it is GA, it looks CAPA needs some code change on it. How should downstream update CAPA, if no new CAPA release comes out when CAPI 1.4 release GA? Or is it possible we can have the new v2.1.0 ready at that time? cc @yastij

Could you raise this in slack channel?

File: controllers/awsmachine_controller_unit_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ankitasw) [reply controllers/awsmachine_controller_unit_test.go:1895]: yes @wyike

**Comment by Skarlso** [comment]
```
0327 14:02:13.049701       1 clusterresourceset_controller.go:339] "failed to apply ClusterResourceSet resource" err="creating object policy/v1beta1, Kind=PodDisruptionBudget kube-system/calico-kube-controllers: no matches for kind \"PodDisruptionBudget\" in version \"policy/v1beta1\"" controller="clusterresourceset" controllerGroup="addons.cluster.x-k8s.io" controllerKind="ClusterResourceSet" ClusterResourceSet="quick-start-nosjme/quick-start-rqkiap-crs-0" namespace="quick-start-nosjme" name="quick-start-rqkiap-crs-0" reconcileID=0729f8d1-defe-4e47-a5fe-126dd09061eb Cluster="quick-start-nosjme/quick-start-rqkiap" Resource kind="ConfigMap" Resource name="cni-quick-start-rqkiap-crs-0"
E0327 14:02:13.060112       1 controller.go:329] "Reconciler error" err="creating object policy/v1beta1, Kind=PodDisruptionBudget kube-system/calico-kube-controllers: no matches for kind \"PodDisruptionBudget\" in version \"policy/v1beta1\"" controller="clusterresourceset" controllerGroup="addons.cluster.x-k8s.io" controllerKind="ClusterResourceSet" ClusterResourceSet="quick-start-nosjme/quick-start-rqkiap-crs-0" namespace="quick-start-nosjme" name="quick-start-rqkiap-crs-0" reconcileID=0729f8d1-defe-4e47-a5fe-126dd09061eb
```

From the CAPI log...

**Comment by Ankitasw** [comment]
PR is ready for review.

cc @richardcase @dlipovetsky if you could take a look?

File: versions.mk [hunk 2]
```diff
@@ -16,6 +16,7 @@ MDBOOK_VERSION := v0.4.5
 PLANTUML_VERSION := 1.2020.16
 GH_VERSION := 2.7.0
 CERT_MANAGER_VERSION := v1.10.1
+CAPI_VERSION := v1.4.0
 CAPI_VERSION := v1.3.5
```
Comment (wyike) [inline versions.mk:20]: do we need to remove line: CAPI_VERSION := v1.3.5

Comment (Ankitasw) [reply versions.mk:20]: thanks for noticing, I will remove

Comment (Ankitasw) [reply versions.mk]: Done

**Comment by Ankitasw** [comment]
Just changed versions.mk so not re-running the e2e tests.

**Comment by Skarlso** [comment]
/lgtm
/approve
/hold
until tests

**Review by yastij — COMMENTED** [review]
/lgtm

**Pull request merged**